videostore/rawsegmenter.c

```diff
@@ -58,71 +47,63 @@ int video_store_raw_seg_init_h264(struct raw_seg_h264 **ppRS, // OUT
   codecCtx = avcodec_alloc_context3(codec);
   if (codecCtx == NULL) {
     av_log(NULL, AV_LOG_ERROR,
-           "video_store_raw_seg_init_h264 failed to allocate codec context");
+           "video_store_raw_seg_init failed to allocate codec context\n");
     ret = VIDEO_STORE_RAW_SEG_RESP_ERROR;
     goto cleanup;
   }
 
   codecCtx->width = width;
   codecCtx->height = height;
-  codecCtx->extradata = av_malloc(extradataSize + AV_INPUT_BUFFER_PADDING_SIZE);
-  if (codecCtx->extradata == NULL) {
-    av_log(NULL, AV_LOG_ERROR,
-           "video_store_raw_seg_init_h264 failed to allocate extradata");
-    ret = VIDEO_STORE_RAW_SEG_RESP_ERROR;
-    goto cleanup;
-  }
-  codecCtx->extradata_size = (int)extradataSize;
 
-  // memcpy can't fail
-  memcpy(codecCtx->extradata, extradata, extradataSize);
-  /* // Copy the codec parameters from the input stream to the output
-   * stream. This is equivalent */
-  /* // to -c:v copy in ffmpeg cli. This is needed to make sure we do not
-   * re-encode the stream. */
   ret = avcodec_parameters_from_context(stream->codecpar, codecCtx);
   if (ret < 0) {
     av_log(NULL, AV_LOG_ERROR,
-           "video_store_raw_seg_init_h264 failed to copy codec parameters");
+           "video_store_raw_seg_init failed to copy codec parameters\n");
     goto cleanup;
   }
+  if (codec->id == AV_CODEC_ID_H265) {
+    // this is needed  to make h265 videos playable on apple devices
+    // https://trac.ffmpeg.org/wiki/Encode/H.265#FinalCutandApplestuffcompatibility
+    // https://stackoverflow.com/questions/50565912/h265-codec-changes-from-hvc1-to-hev1
+    stream->codecpar->codec_tag = MKTAG('h', 'v', 'c', '1');
```

Comment:
I've confirmed that the videos (both h264 andf h265) produced by the concater are playable on mac by vlc and quicktime, and on android